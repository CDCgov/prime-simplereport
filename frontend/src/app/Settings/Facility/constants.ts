const isRequiredErrMsg = (fieldIdentifier: string) => {
  return `${fieldIdentifier} is required`;
};

const isInvalidErrMsg = (fieldIdentifier: string) => {
  return `${fieldIdentifier} is invalid`;
};

const facilityInfoLabels = {
  name: "Facility name",
  phone: "Facility phone number",
  email: "Email",
  street: "Facility street",
  zip: "Facility ZIP code",
  state: "Facility state",
  clia: "Facility CLIA number",
};

const orderingProviderLabels = {
  first: "Ordering provider first name",
  last: "Ordering provider last name",
  phone: "Ordering provider phone number",
  zip: "Ordering provider ZIP code",
};

export const facilityInfoErrMsgs = {
  name: {
    required: isRequiredErrMsg(facilityInfoLabels.name),
  },
  phone: {
    required: isRequiredErrMsg(facilityInfoLabels.phone),
    invalid: isInvalidErrMsg(facilityInfoLabels.phone),
  },
  email: {
    invalid: `${facilityInfoLabels.email} is incorrectly formatted`,
  },
  street: {
    required: isRequiredErrMsg(facilityInfoLabels.street),
  },
  zip: {
    required: isRequiredErrMsg(facilityInfoLabels.zip),
    invalid: isInvalidErrMsg(facilityInfoLabels.zip),
    stateInvalid: "Invalid ZIP code for this state",
  },
  state: {
    required: isRequiredErrMsg(facilityInfoLabels.state),
    invalid: "SimpleReport isnâ€™t currently supported in",
  },
  clia: {
    required: isRequiredErrMsg(facilityInfoLabels.clia),
    invalid:
      "CLIA numbers must be 10 characters (##D#######), or a special temporary number from CA, IL, VT, WA, WY, or the Department of Defense",
  },
};

export const orderingProviderErrMsgs = {
  first: {
    required: isRequiredErrMsg(orderingProviderLabels.first),
  },
  last: {
    required: isRequiredErrMsg(orderingProviderLabels.last),
  },
  npi: {
    required: "NPI should be a 10-digit numerical value (##########)",
  },
  phone: {
    required: isRequiredErrMsg(orderingProviderLabels.phone),
    invalid: isInvalidErrMsg(orderingProviderLabels.phone),
  },
  zip: {
    invalid: isInvalidErrMsg(orderingProviderLabels.zip),
  },
};

export const deviceRequiredErrMsg = "There must be at least one device";
