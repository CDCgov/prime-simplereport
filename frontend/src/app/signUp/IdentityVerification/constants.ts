export const exampleQuestionSet = [
  {
    questionType: 28,
    questionText:
      "Please select the model year of the vehicle you purchased or leased prior to January 2011 .",
    questionSelect: {
      questionChoice: [
        "2002",
        "2003",
        "2004",
        "2005",
        "NONE OF THE ABOVE/DOES NOT APPLY",
      ],
    },
  },
  {
    questionType: 24,
    questionText:
      "Which of the following professions do you currently have, or if there is not a matched profession, please select 'NONE OF THE ABOVE'.",
    questionSelect: {
      questionChoice: [
        "DENTIST / DENTAL HYGIENIST",
        "SOCIAL WORKER",
        "OPTICIAN / OPTOMETRIST",
        "ELECTRICIAN",
        "NONE OF THE ABOVE/DOES NOT APPLY",
      ],
    },
  },
  {
    questionType: 41,
    questionText:
      "Please select the number of bedrooms in your home from the following choices. If the number of bedrooms in your home is not one of the choices please select 'NONE OF THE ABOVE'.",
    questionSelect: {
      questionChoice: [
        "TWO",
        "THREE",
        "FOUR",
        "FIVE",
        "NONE OF THE ABOVE/DOES NOT APPLY",
      ],
    },
  },
  {
    questionType: 47,
    questionText:
      "According to your credit profile, you may have opened a Home type loan in or around May 2015. Please select the lender to whom you currently made your payments.",
    questionSelect: {
      questionChoice: [
        "UC LENDING",
        "CTX MORTGAGE",
        "MID AMERICA MORTGAGE",
        "1ST NATIONWIDE MTG",
        "NONE OF THE ABOVE/DOES NOT APPLY",
      ],
    },
  },
  {
    questionType: 49,
    questionText:
      "According to our records, you graduated from which of the following High Schools?",
    questionSelect: {
      questionChoice: [
        "BAXTER SPRINGS HIGH SCHOOL",
        "BELLS HIGH SCHOOL",
        "LOCKNEY HIGH SCHOOL",
        "AGUA DULCE HIGH SCHOOL",
        "NONE OF THE ABOVE/DOES NOT APPLY",
      ],
    },
  },
];
