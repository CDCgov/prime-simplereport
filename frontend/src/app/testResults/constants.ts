export enum MULTIPLEX_DISEASES {
  COVID_19 = "COVID-19",
  FLU_A = "Flu A",
  FLU_B = "Flu B",
  RSV = "RSV",
  HIV = "HIV",
  SYPHILIS = "Syphilis",
  FLU_A_AND_B = "Flu A and B",
}

export enum TEST_RESULTS {
  POSITIVE = "POSITIVE",
  NEGATIVE = "NEGATIVE",
  UNDETERMINED = "UNDETERMINED",
  UNKNOWN = "UNKNOWN",
}
