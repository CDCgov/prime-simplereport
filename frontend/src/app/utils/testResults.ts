function getTestResult(result: MultiplexResult): TestResult {
  if (result) {
    if ("testResult" in result) {
      return result.testResult;
    }
  }
  return "UNKNOWN";
}

export function getResultObjByDiseaseName(
  results: MultiplexResults,
  diseaseName: MultiplexDisease
): MultiplexResult {
  return (
    (results.find((result: MultiplexResult) => {
      return result.disease.name.includes(diseaseName);
    }) as MultiplexResult) || null
  );
}

export function getResultByDiseaseName(
  results: MultiplexResults,
  diseaseName: MultiplexDisease
): string {
  const result = getResultObjByDiseaseName(results, diseaseName);
  return getTestResult(result) || "UNKNOWN";
}

export function getSortedResults(results: MultiplexResults): MultiplexResults {
  return Object.values(results).sort(
    (a: MultiplexResult, b: MultiplexResult) => {
      return a.disease.name.localeCompare(b.disease.name);
    }
  );
}

export function hasMultiplexResults(results: MultiplexResults): boolean {
  return results?.length > 1
    ? results.some(
        (result: MultiplexResult) => result.disease?.name !== "COVID-19"
      )
    : false;
}

export function hasPositiveFluResults(results: MultiplexResults): boolean {
  return (
    results.filter(
      (multiplexResult: MultiplexResult) =>
        multiplexResult.disease.name.includes("Flu") &&
        getTestResult(multiplexResult) === "POSITIVE"
    ).length > 0
  );
}
