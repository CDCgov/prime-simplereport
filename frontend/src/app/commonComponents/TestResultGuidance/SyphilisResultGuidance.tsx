import React from "react";
import { Trans, useTranslation } from "react-i18next";

import { TEST_RESULTS } from "../../testResults/constants";

import { ResultGuidanceProps } from "./ResultsGuidance";

const PositiveSyphilisResultInfo = () => {
  const { t } = useTranslation();

  return (
    <>
      <p className="text-bold sr-guidance-heading">
        {t("testResult.syphilisNotes.h1")}
      </p>
      <p>{t("testResult.syphilisNotes.positive.p0")}</p>
      <Trans
        parent="p"
        i18nKey="testResult.syphilisNotes.positive.p1"
        components={[
          <a
            href={t("testResult.syphilisNotes.positive.treatmentLink")}
            target="_blank"
            rel="noopener noreferrer"
            key="syphilis-treatment-link"
          >
            syphilis treatment link
          </a>,
        ]}
      />
    </>
  );
};

const SyphilisResultGuidance = ({ result }: ResultGuidanceProps) => {
  const hasPositiveSyphilisResult = result.testResult === TEST_RESULTS.POSITIVE;
  return <>{hasPositiveSyphilisResult && <PositiveSyphilisResultInfo />}</>;
};

export default SyphilisResultGuidance;
