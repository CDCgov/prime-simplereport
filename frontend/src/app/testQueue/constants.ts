import { AlertContent } from "../commonComponents/Alert";
import { displayFullName } from "../utils";

export const QUEUE_NOTIFICATION_TYPES = {
  ADDED_TO_QUEUE__SUCCESS: 1,
  SUBMITTED_RESULT__SUCCESS: 2,
  DELIVERED_RESULT_TO_PATIENT__FAILURE: 3,
};

export type SomeoneWithName = {
  firstName?: string | null;
  middleName?: string | null;
  lastName?: string | null;
};

export const ALERT_CONTENT = {
  [QUEUE_NOTIFICATION_TYPES.ADDED_TO_QUEUE__SUCCESS]: <
    T extends SomeoneWithName
  >(
    patient: T,
    startWithLastName: boolean = true
  ): AlertContent => {
    return {
      type: "success",
      title: `${displayFullName(
        patient.firstName,
        patient.middleName,
        patient.lastName,
        startWithLastName
      )} was added to the queue`,
      body: "Newly added patients go to the bottom of the queue",
    };
  },
  [QUEUE_NOTIFICATION_TYPES.SUBMITTED_RESULT__SUCCESS]: <
    T extends SomeoneWithName
  >(
    patient: T,
    startWithLastName: boolean = true,
    dataRetentionLimitsEnabled: boolean = false
  ): AlertContent => {
    const resultText = dataRetentionLimitsEnabled
      ? "has been sent."
      : "was saved and reported.";
    return {
      type: "success",
      title: `Result for ${displayFullName(
        patient.firstName,
        patient.middleName,
        patient.lastName,
        startWithLastName
      )} ${resultText}`,
      body: "See Results to view all test submissions",
    };
  },
  [QUEUE_NOTIFICATION_TYPES.DELIVERED_RESULT_TO_PATIENT__FAILURE]: <
    T extends SomeoneWithName
  >(
    patient: T,
    startWithLastName: boolean = true
  ): AlertContent => {
    return {
      type: "error",
      title: `Unable to text result to ${displayFullName(
        patient.firstName,
        patient.middleName,
        patient.lastName,
        startWithLastName
      )}`,
      body: "The phone number provided may not be valid or may not be able to accept text messages",
    };
  },
};

export const MIN_SEARCH_CHARACTER_COUNT = 2;

export const SEARCH_DEBOUNCE_TIME = 500;
