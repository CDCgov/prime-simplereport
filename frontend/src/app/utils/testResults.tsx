import { MULTIPLEX_DISEASES, TEST_RESULTS } from "../testResults/constants";
import { MultiplexResultInput } from "../../generated/graphql";
import CovidResultGuidance from "../commonComponents/TestResultGuidance/CovidResultGuidance";
import FluResultGuidance from "../commonComponents/TestResultGuidance/FluResultGuidance";
import HivResultGuidance from "../commonComponents/TestResultGuidance/HivResultGuidance";
import RsvResultGuidance from "../commonComponents/TestResultGuidance/RsvResultGuidance";
import SyphilisResultGuidance from "../commonComponents/TestResultGuidance/SyphilisResultGuidance";
import HepatitisCResultGuidance from "../commonComponents/TestResultGuidance/HepatitisCResultGuidance";
import GonorrheaResultGuidance from "../commonComponents/TestResultGuidance/GonorrheaResultGuidance";
import ChlamydiaResultGuidance from "../commonComponents/TestResultGuidance/ChlamydiaResultGuidance";

export function getResultByDiseaseName(
  results: MultiplexResults,
  diseaseName: MultiplexDisease
): string;
export function getResultByDiseaseName(
  results: MultiplexResultInput[],
  diseaseName: MULTIPLEX_DISEASES
): string;
export function getResultByDiseaseName(
  results: MultiplexResults | MultiplexResultInput[],
  diseaseName: MultiplexDisease | MULTIPLEX_DISEASES
): string {
  const matchedResult = results.find((r) => {
    if (isMultiplexInput(r)) {
      return r.diseaseName === diseaseName;
    } else {
      return r.disease.name.includes(diseaseName);
    }
  });

  if (matchedResult === undefined) return "UNKNOWN";
  return matchedResult.testResult;
}

function isMultiplexInput(
  result: MultiplexResult | MultiplexResultInput
): result is MultiplexResultInput {
  return (result as MultiplexResultInput).diseaseName !== undefined;
}

export function getSortedResults(results: MultiplexResults): MultiplexResults {
  return Object.values(results).sort(
    (a: MultiplexResult, b: MultiplexResult) => {
      return a.disease.name.localeCompare(b.disease.name);
    }
  );
}

export function hasMultipleResults(results: MultiplexResults): boolean {
  return results?.length > 1;
}

export function getResultForDisease(
  results: MultiplexResults | null | undefined,
  diseaseName: MultiplexDisease,
  positivesOnly: boolean = false
): MultiplexResult | undefined {
  if (!results) {
    return undefined;
  }
  const matches: MultiplexResult[] = results.filter(
    (multiplexResult: MultiplexResult) =>
      multiplexResult.disease.name === diseaseName &&
      (!positivesOnly || multiplexResult.testResult === TEST_RESULTS.POSITIVE)
  );
  return matches.at(0);
}

export const getGuidanceForResults = (
  results: MultiplexResult[],
  isPatientApp: boolean
) => {
  const guidance: JSX.Element[] = [];

  let match = getResultForDisease(results, MULTIPLEX_DISEASES.COVID_19);
  if (match) {
    guidance.push(
      <CovidResultGuidance result={match} isPatientApp={isPatientApp} />
    );
  }

  match =
    getResultForDisease(results, MULTIPLEX_DISEASES.FLU_A, true) ||
    getResultForDisease(results, MULTIPLEX_DISEASES.FLU_B, true) ||
    getResultForDisease(results, MULTIPLEX_DISEASES.FLU_A_AND_B, true);
  if (match) {
    guidance.push(<FluResultGuidance result={match} />);
  }

  match = getResultForDisease(results, MULTIPLEX_DISEASES.HIV);
  if (match) {
    guidance.push(<HivResultGuidance />);
  }

  match = getResultForDisease(results, MULTIPLEX_DISEASES.RSV, true);
  if (match) {
    guidance.push(<RsvResultGuidance result={match} />);
  }

  match = getResultForDisease(results, MULTIPLEX_DISEASES.SYPHILIS, true);
  if (match) {
    guidance.push(<SyphilisResultGuidance result={match} />);
  }

  match = getResultForDisease(results, MULTIPLEX_DISEASES.HEPATITIS_C);
  if (match) {
    guidance.push(<HepatitisCResultGuidance />);
  }

  match = getResultForDisease(results, MULTIPLEX_DISEASES.GONORRHEA);
  if (match) {
    guidance.push(<GonorrheaResultGuidance />);
  }

  match = getResultForDisease(results, MULTIPLEX_DISEASES.CHLAMYDIA);
  if (match) {
    guidance.push(<ChlamydiaResultGuidance />);
  }

  return guidance;
};
