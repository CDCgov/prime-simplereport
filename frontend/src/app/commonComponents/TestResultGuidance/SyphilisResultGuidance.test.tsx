import { render, screen } from "@testing-library/react";
import { cloneDeep } from "lodash";

import "../../../i18n";

import { MULTIPLEX_DISEASES, TEST_RESULTS } from "../../testResults/constants";

import SyphilisResultGuidance from "./SyphilisResultGuidance";

describe("SyphilisResultGuidance", () => {
  const mockResultTemplate = {
    disease: {
      name: MULTIPLEX_DISEASES.SYPHILIS,
    },
    testResult: TEST_RESULTS.POSITIVE,
  } as MultiplexResult;

  it("displays guidance for a positive RSV result", () => {
    const { container } = render(
      <SyphilisResultGuidance result={mockResultTemplate} />
    );
    expect(container).toMatchSnapshot();
  });

  const nonPositiveCases = [
    TEST_RESULTS.NEGATIVE,
    TEST_RESULTS.UNDETERMINED,
    TEST_RESULTS.UNKNOWN,
  ];

  test.each(nonPositiveCases)(
    "displays no RSV guidance for %p result",
    (testResult) => {
      const mockResult = cloneDeep(mockResultTemplate);
      mockResult.testResult = testResult;
      render(<SyphilisResultGuidance result={mockResult} />);
      expect(screen.queryByText("For Syphilis")).not.toBeInTheDocument();
    }
  );
});
