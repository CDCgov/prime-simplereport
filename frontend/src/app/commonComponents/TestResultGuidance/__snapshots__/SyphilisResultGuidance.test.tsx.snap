// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyphilisResultGuidance displays guidance for a positive syphilis result 1`] = `
<div>
  <p
    class="text-bold sr-guidance-heading"
  >
    For syphilis:
  </p>
  <p>
    If you have a positive result, you will need a follow-up test to confirm your results. The organization that provided your test should be able to answer questions and provide referrals for follow-up testing.
  </p>
  <p>
    <a
      href="https://www.cdc.gov/std/treatment-guidelines/syphilis.htm"
      rel="noopener noreferrer"
      target="_blank"
    >
      Visit the CDC website to learn more about a positive Syphilis result
    </a>
     (cdc.gov/std/treatment-guidelines/syphilis.htm).
  </p>
</div>
`;
